src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -119,26 +132,43 @@ def __init__(
             asyncio.CancelledError: when the task gets cancelled.
         """
         assert (
-            sampling_period.total_seconds() > 0
-        ), "The sampling period should be greater than zero."
+            input_sampling_period.total_seconds() > 0
+        ), "The input sampling period should be greater than zero."
         assert (
-            sampling_period <= size
-        ), "The sampling period should be equal to or lower than the window size."
+            input_sampling_period <= size
+        ), "The input sampling period should be equal to or lower than the window size."
+
+        sampling = input_sampling_period
+        self._resampler: Resampler | None = None
+        self._avg_sender: Sender[Sample] | None = None
```

Comment:
This `avg_sender` got me really confused. This is just the channel where to send samples to get resampled by the resampler, right? If so, I would name it `resampling_source` or `resampling_sender` to make it more clear it only exists for resampling.

Reply:
You're right, `resampling_sender` sounds good to me

Reply:
Actually it seems it might be better to call it `resampler_sender` to be consistent with the other names e.g. resampler_config, resampler_task, etc